src/socket.rs

```diff
@@ -118,11 +118,8 @@ impl Socket {
     ///
     /// This function sets the same flags as in done for [`Socket::new`],
     /// [`Socket::pair_raw`] can be used if you don't want to set those flags.
-    ///
-    /// # Notes
-    ///
-    /// This function is only available on Unix.
-    #[cfg(all(feature = "all", unix))]
+    #[cfg(any(doc, all(feature = "all", unix)))]
```

Comment:
I don't think the `any(doc, ...` part works., same for for a number of other functions.

Reply:
Same as above - this will allow this function to appear in documentation generated on a Windows machine (rustdoc doesn't check function bodies so its OK). So someone will not have to check every platform's docs to see that this function exists.